src/clockkit.cs

```diff
@@ -200,6 +304,15 @@ interface CLKComplicationTemplateModularSmallColumnsText {
 
 		[Export ("highlightColumn2")]
 		bool HighlightColumn2 { get; set; }
+
+		[Watch (7, 0), NoiOS]
+		[Export ("initWithRow1Column1TextProvider:row1Column2TextProvider:row2Column1TextProvider:row2Column2TextProvider:")]
+		IntPtr Constructor (CLKTextProvider row1Column1TextProvider, CLKTextProvider row1Column2TextProvider, CLKTextProvider row2Column1TextProvider, CLKTextProvider row2Column2TextProvider);
```

Comment:
I wonder whether you could simplify all of these param names to row1Column1Provider since they're all text providers?

Reply:
Open to suggestions, but the name has to point to the row and the column, we could remove TextProvider, but doesn't make it much nicer :/ 

As rolf said, someone does not know about arrays

Reply:
A custom `.ctor (CLKTextProvider[,] textProviders)` with additional validations (ranges) could be _manually_ added (and tested) and this one could become internal (or protected for subclassing).

Reply:
You might know a trick I don't with C# because the way to call another constructor on c# is:
```
public CLKComplicationTemplateModularLargeTable (CLKTextProvider[] data) : this (data[0], data[1], data[2], data[3], data[4]) {
}
```
And this code is super error prone. I cannot validate that data is not null AND I cannot check the number of items before I call this. Factory methods, Ok, no problem, the constructors, I don't see how unless I do a binding to a IntPtr for the objc msg send etc..

Reply:
There's an ugly trick using a static method `: this (CheckData (data), data [1], ...)`
But better call objc_msgSend (and not `this`)
Not sure it's worth it... at least _right now_ (can be added at any time)